### ISSUE TITLE
Rewrite relative links and images in README

### ISSUE THREAD

**Issue by chillu** [silverstripe/addons.silverstripe.org#50]
Special case for github: Prefix all relative links with `http://raw.github.com` and their repo path, to avoid broken images. 

Or see if we can use the Github rendering API with the `context` attr for it: http://developer.github.com/v3/markdown/

Example: http://addons.silverstripe.org/add-ons/heyday/silverstripe-dataobjectpreview

**Comment by sminnee** [comment]
Another example: http://addons.silverstripe.org/add-ons/colymba/gridfield-bulk-editing-tools

It affects both image embeds and links.

**Comment by robbieaverill** [comment]
This actually could be part of a slightly bigger picture, in that we probably need to handle vanilla markdown and GitHub markdown separately. If the GitHub context was already established then we can add the image and link prefixes for relative paths, as well as probably using GitHub's markdown parser without needing to many "ifs" and "elses" etc.

**Comment by chillu** [comment]
Isn't Github markdown a superset of Markdown? I'd assume that we can just pass all README.md through this API, and set a context if the repo URL is `github.com`

**Comment by robbieaverill** [comment]
Yeah OK, I'm fine with that. I'll adjust #142 tomorrow with a GitHub parser.

**Comment by robbieaverill** [comment]
@chillu do you think it's worth porting the custom filters from docs.ss to somewhere central so it can be used again here for consistency?

**Comment by chillu** [comment]
You mean the `[api:SomeClass]` stuff? It would be nice to have that extracted, although it's not essential. Where would you see it used outside of doc.ss.org? On a somewhat related note, I think we're spending too much time on maintaining doc.ss.org, wish we would've just gone with https://pages.github.com/ or some other static page generator a few years ago :)

**Comment by robbieaverill** [comment]
PR for using the GitHub API available in #142 - I'll start look at fixing the relative URLs now

**Comment by robbieaverill** [comment]
I've covered this change in [this commit](https://github.com/robbieaverill/addons.silverstripe.org/commit/5cc7d1e7fd5197e22e89245ffaa6d3accf2374b9). I'll put a pull request in for it once #142 is merged since the branch contains one of the same commits.
